design/events.md

```diff
@@ -0,0 +1,439 @@
+				</ul>
+			</td>
+			<td>Operations are listed in summarized form (only keys without contents), in order to avoid huge event payloads. 
+          Full operation contents are sent separately in content-related events.
+				<p>A <code>COMMIT</code> event should trigger:</p>
+				<ul>
+					<li>0-N content events (one for each PUT/DELETE operation).</li>
+				</ul>
+			</td>
+		</tr>
+		<tr>
+			<td><code>MERGE</code></td>
+			<td>API call</td>
+			<td>
+				<ul>
+					<li>Source branch name</li>
+					<li>Target branch name</li>
+					<li>Previous target branch HEAD</li>
+					<li>New target branch HEAD</li>
+					<li>Common ancestor hash</li>
+					<li>List of commits with their operations (list of PUTs and DELETEs)</li>
+				</ul>
+			</td>
+			<td>Operations are listed in summarized form (only keys without contents), in order to avoid huge event payloads. 
+          Full operation contents are sent separately in content-related events.
+				<p>A <code>MERGE</code> event should trigger:</p>
+				<ul>
+					<li>0-N content events (one for each global PUT/DELETE operation, excluding intermediary operations).</li>
+				</ul>
+			</td>
+		</tr>
+		<tr>
+			<td><code>TRANSPLANT</code></td>
+			<td>API call</td>
+			<td>
+				<ul>
+					<li>Source branch name</li>
+					<li>Target branch name</li>
+					<li>Previous target branch HEAD</li>
+					<li>New target branch HEAD</li>
+					<li>List of commits with their operations (list of PUTs and DELETEs)</li>
+				</ul>
+			</td>
+			<td>Operations are listed in summarized form (only keys without contents), in order to avoid huge event payloads. 
+          Full operation contents are sent separately in content-related events.
+				<p>A <code>TRANSPLANT</code> event should trigger:</p>
+				<ul>
+					<li>0-N content events (one for each global PUT/DELETE operation, excluding intermediary operations).</li>
+				</ul>
+			</td>
+		</tr>
+		<tr>
+			<td colspan="2" rowspan="2">Content-related events</td>
+			<td><code>CONTENT_STORED</code></td>
+			<td>Committing event</td>
+			<td>
+				<ul>
+					<li>Branch name</li>
+					<li>Commit Hash</li>
+					<li>Content type</li>
+					<li>Content key</li>
+					<li><code>Content</code> object</li>
+				</ul>
+			</td>
+			<td>Corresponds to a PUT operation from a committing event. Note that it's not possible to distinguish a creation 
+          from an update.</td>
+		</tr>
+		<tr>
+			<td><code>CONTENT_REMOVED</code></td>
+			<td>Committing event</td>
+			<td>
+				<ul>
+					<li>Branch name</li>
+					<li>Commit Hash</li>
+					<li>Content type</li>
+					<li>Content key</li>
+				</ul>
+			</td>
+			<td>Corresponds to a DELETE operation from a committing event. Note that the actual Content object that was 
+          deleted is not available.</td>
+		</tr>
+	</tbody>
+</table>
+
+### 2.1.5. Common event attributes
+
+Every event should carry the following contextual attributes:
+
+* Event ID: a (random) UUID that uniquely identifies the event.
```

Comment:
> No "Event type" as a common attribute?

Sorry, that was an oversight, there will be of course an event type field.

> For event types in general, can you add a note that consumers must protect themselves against added event types (aka: no enum or enum with a fallback UNKNOWN value)?

Good idea, will do.

Reply:
Although, a new event type would necessarily come with a new `EventListener` method, so there is little risk that an existing consumer would break because of a new event type. But your remark remains very relevant.